src/CakeSlayer/Has.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+{- | This module introduces the type class 'Has'.
+-}
+
+module CakeSlayer.Has
+       ( Has (..)
+       , grab
+       ) where
+
+
+{- | General type class representing which @field@ is in @env@.
```

Comment:
Should the documentation also mention how to define instances of the `Has` typeclass? Or is it obvious?

Reply:
Yes, makes sense!